server/user.js

```diff
@@ -0,0 +1,85 @@
+const { TimeLogger } = require("../src/util");
+const { R } = require("redbean-node");
+const { UptimeKumaServer } = require("./uptime-kuma-server");
+const server = UptimeKumaServer.getInstance();
+const io = server.io;
+
+/**
+ * Send list of users to client
+ * @param {Socket} socket Socket.io socket instance
+ * @returns {Promise<Bean[]>}
+ */
+async function sendUserList(socket) {
+    const timeLogger = new TimeLogger();
+
+    let result = [];
+    let list = (await R.findAll("user")).map(({ id, username, active }) => ({
+        id,
+        username,
+        active
+    }));
+
+    io.to(socket.userID).emit("userList", result);
+
+    timeLogger.print("Send User List");
+
+    return list;
+}
+
+/**
+ * Send list of users to client
+ * @param {Socket} socket Socket.io socket instance
+ * @returns {Promise<Bean[]>}
+ */
+async function getUser(userID) {
+    const timeLogger = new TimeLogger();
+
+    const bean = await R.findOne("user", " id = ? ", [ userID ])
+        .then(({ id, username, active }) => ({
+            id,
+            username,
+            active
+        }));
+
+    if (!bean) {
+        throw new Error("User not found");
+    }
+
+    timeLogger.print(`Get user ${userID}`);
+
+    return bean;
+}
+
+/**
+ * Saves and updates given user entity
+ * @param {Socket} socket Socket.io socket instance
+ * @param user
+ * @returns {Promise<Bean[]>}
```

Comment:
see reasoning below

```suggestion
 * @returns {Promise<void>}
```

Reply:
You are right it's useless and don't return it allow me to remove the `.map` used to keep control over returned fields. In fact I'm usually writing GraphQL API, where I have to return the updated object as mutation result (used by the webapp to refresh his cache).